### PR TITLE
Docs: rewriting the "Setting options" topic

### PR DESCRIPTION
**Pull request by kirszenbaum**
This PR:
- Rewrites the [Setting options](https://handsontable.com/docs/setting-options/) topic:
http://localhost:8080/docs/next/setting-options/

The topic is now called "Configuration options" (similarly to the "Configuration options" section in the [HyperFormula docs](https://handsontable.github.io/hyperformula/guide/configuration-options.html)).

Any content repetitions are by design (each section of the topic should be able to function as a stand-alone procedure).

In the next step, the [Options](https://handsontable.com/docs/api/options/) section of the API ref will be rewritten, in consistency with this PR's changes.

[skip changelog]

### CODE REVIEW

File: docs/next/guides/getting-started/setting-options.md [hunk 1]
```diff
@@ -1,239 +1,674 @@
...
-hot.getCellMeta(1, 1).readOnly === true; // the "cells" overwrites the value
+// for cell (1, 1), the `cell` option overwrote the default `readOnly` value
+// returns `true`
+hot.getCellMeta(1, 1).readOnly;
 
-// In this case, an attempt to change the state will have no effect. The `readOnly`
-// property always returns `true`.
-hot.setCellMeta(1, 1, 'readOnly', false);
+// change the `readOnly` option of cell (1, 1) back to `false`
+hot.setCellMeta(1, 1, readOnly, false);
```
Comment (AMBudnik) [inline docs/next/guides/getting-started/setting-options.md:543]: > hot.setCellMeta(1, 1, readOnly, true);

that will not work, `readOnly` has to be a string. Please change it to

> hot.setCellMeta(1, 1, 'readOnly', true);

File: docs/next/guides/getting-started/setting-options.md [hunk 2]
```diff
@@ -1,239 +1,674 @@
...
+```
+:::
+
+## Configuration example
+
+In the example below, some cells are read-only, and some cells are editable:
+- By default, all cells are read-only (as set in the top-level [grid options](#setting-grid-options)).
+- For the first column, the mid-level [column options](#setting-column-options) overwrite the top-level [grid options](#setting-grid-options).<br>
+  As a result, the first column cells are editable.
+- For cell `A1` (`0`, `0`), the bottom-level [cell options](#setting-cell-options) overwrite both the mid-level [column options](#setting-column-options), and the top-level [grid options.](#setting-grid-options)<br>
```
Comment (AMBudnik) [inline docs/next/guides/getting-started/setting-options.md:641]: Here is the same situation as the one above we could change the 

> For cell `A1` (`0`, `0`),

to

> For cell `A1` `(0, 0)`,

it is definitely easier to read

**Review by AMBudnik — CHANGES_REQUESTED** [review]
Hey @kirszenbaum 

Thanks for adding me as a reviewer. I added some suggestions within the code during the review. I also have some general thoughts

* we mention `setCellMeta`  after using it a couple of times before in examples
* we do not mention that `setCellMeta`  does not generate the alteration of the `readOnly` mode till the cell is rerendered. Fortunately, `getCellMeta`  does not need a render so it's ok.
* the `implementing custom logic` example seems to be pretty much the same as the one above. If we want to have this paragraph maybe we should use other options and/or play with conditions. Or maybe add some custom cellmeta (example https://jsfiddle.net/handsoncode/tb7rw0xk/) 

Maybe I would also add something about the options which while added change the default set of keys in the dropdown and context menus. But it's only a thought. Maybe the context menu tutorial section is a better place.

File: docs/next/guides/getting-started/setting-options.md [hunk 3]
```diff
@@ -1,239 +1,674 @@
...
 });
+
+// check a cell's options
+// returns `true`
+hot.getCellMeta(0, 0).readOnly;
 ```
 :::
 
-### Single column
+## Setting column options
```
[the remark this answers is not in the document]
Comment (budnix) [reply docs/next/guides/getting-started/setting-options.md:137]: The `hight` and `width` options prevent the table to not jump out from the parent element. Related to https://github.com/handsontable/handsontable/pull/8449.

[the remark this answers is not in the document]
Comment (AMBudnik) [reply docs/next/guides/getting-started/setting-options.md:137]: Thanks @budnix. It seems like we need it then.
But I still would like to change the declaration of columns from `column:` to `columns: []` (an empty array)

**Review by AMBudnik — APPROVED** [review]
@kirszenbaum, thank you for addressing all of my tips

File: docs/next/guides/getting-started/setting-options.md [hunk 4]
```diff
@@ -1,239 +1,631 @@
...
-      className: 'bg-read-only'
-    },
-    {},
-  ],
   rowHeaders: true,
   colHeaders: true,
-  licenseKey: 'non-commercial-and-evaluation'
+  // in the top-level grid options, all cells are editable
+  readOnly: false,
+  columns: function(index) {
```
Comment (swistach) [inline docs/next/guides/getting-started/setting-options.md:214]:
```suggestion
  columns(index) {
```

File: docs/next/guides/getting-started/setting-options.md [hunk 5]
```diff
@@ -1,239 +1,631 @@
...
+    return {
+      type: index > 0 ? 'numeric' : 'text',
+      readOnly: index == 2 || index == 8
```
Comment (swistach) [inline docs/next/guides/getting-started/setting-options.md:217]:
```suggestion
      readOnly: index === 2 || index === 8
```

File: docs/next/guides/getting-started/setting-options.md [hunk 6]
```diff
@@ -1,239 +1,631 @@
...
+:::
+
+1. Within [Handsontable constructor](@/guides/getting-started/installation.md#initialize-the-grid)'s second argument, add an option called [`cells`](@/api/options.md#cells), and set it to a function.
+    ```js
+    const hot = new Handsontable(container, {
+      // top-level grid options that apply to the entire grid
+      width: 400,
+      height: 300,
+      // the `cells` option
+      cells: function() {
```
Comment (swistach) [inline docs/next/guides/getting-started/setting-options.md:248]:
```suggestion
      cells() {
```

File: docs/next/guides/getting-started/setting-options.md [hunk 7]
```diff
@@ -1,239 +1,631 @@
...
+    ```
+2. The function can take three arguments:<br>
+   - `row`: a row coordinate (a **physical** index)
+   - `col`: a column coordinate (a **physical** index)
+   - `prop`: if your [`data`](@/api/options.md#data) is an [array of objects](@/guides/getting-started/binding-to-data.md#array-of-objects), `prop` is a property name for a column's data source object.<br>
+   If your [`data`](@/api/options.md#data) is an [array of arrays](@/guides/getting-started/binding-to-data.md#array-of-arrays), `prop` is the same as `col`.
+    ```js
+    const hot = new Handsontable(container, {
+      // the `cells` option set to a function
+      cells: function(row, col, prop) {
```
Comment (swistach) [inline docs/next/guides/getting-started/setting-options.md:261]:
```suggestion
      cells(row, col, prop) {
```

File: docs/next/guides/getting-started/setting-options.md [hunk 8]
```diff
@@ -1,239 +1,631 @@
...
+        // the `cells` function's body
+      }
+    });
+    ```
+3. In the [`cells`](@/api/options.md#cells) function's body, implement a logic that selects your required row(s).<br>
+   For example, to make each cell of the first row and each cell of the fourth row read-only:
+    ```js
+    const hot = new Handsontable(container, {
+      // the `cells` options overwrite all other options
+      cells: function(row, col, prop) {
```
Comment (swistach) [inline docs/next/guides/getting-started/setting-options.md:271]:
```suggestion
      cells(row, col, prop) {
```

File: docs/next/guides/getting-started/setting-options.md [hunk 9]
```diff
@@ -1,239 +1,631 @@
...
+  licenseKey: 'non-commercial-and-evaluation',
   data: Handsontable.helper.createSpreadsheetData(5, 10),
-  cells(row, column) {
+  width: 'auto',
+  height: 'auto',
+  rowHeaders: true,
+  colHeaders: true,
+  // the `cells` options overwrite all other options
+  // apply only to each cell of rows 1 and 4, as specified in the function's body
+  cells: function(row, column) {
```
Comment (swistach) [inline docs/next/guides/getting-started/setting-options.md:310]:
```suggestion
  cells(row, column) {
```

File: docs/next/guides/getting-started/setting-options.md [hunk 10]
```diff
@@ -1,239 +1,631 @@
...
 
-## `cells` vs `cell`
+### Checking cell options
 
-In the Options section of the API, you can find two similar options, [cell](@/api/options.md#cells-2) and [cell](@/api/options.md#cell). Using one of these options (or both), you can associate the custom properties or overwrite an entire table or column option for a particular cell. Both options allow you to set initial values for specific cells. You can check if the values are propagated by calling the [getCellMeta](@/api/core.md#getcellmeta) method. The method returns an object with all built-in options as well as those added by the developer.
+When Handsontable is running, you can check a cell's current options, using the [`getCellMeta()`](@/api/core.md#getcellmeta) method.
 
-#### `cell`
+The [`getCellMeta()`](@/api/core.md#getcellmeta) method returns an object with:
+- All built-in options
```
Comment (swistach) [inline docs/next/guides/getting-started/setting-options.md:434]: We can mention here that built-in properties are stored in the `prototype` of CellMeta (https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Objects/Object_prototypes).

File: docs/next/guides/getting-started/setting-options.md [hunk 11]
```diff
@@ -1,239 +1,631 @@
...
+:::
+
+1. Within [Handsontable constructor](@/guides/getting-started/installation.md#initialize-the-grid)'s second argument, add an option called [`cells`](@/api/cells.md#cells), and set it to a function.
+    ```js
+    const hot = new Handsontable(container, {
+      // top-level grid options that apply to the entire grid
+      width: 400,
+      height: 300,
+      // the `cells` option
+      cells: function() {
```
Comment (swistach) [inline docs/next/guides/getting-started/setting-options.md:534]:
```suggestion
      cells() {
```

File: docs/next/guides/getting-started/setting-options.md [hunk 12]
```diff
@@ -1,239 +1,631 @@
...
+    ```
+2. The function can take three arguments:<br>
+   - `row`: a row coordinate (a **physical** index)
+   - `col`: a column coordinate (a **physical** index)
+   - `prop`: if your [`data`](@/api/options.md#data) is an [array of objects](@/guides/getting-started/binding-to-data.md#array-of-objects), `prop` is a property name for a column's data source object.<br>
+   If your [`data`](@/api/options.md#data) is an [array of arrays](@/guides/getting-started/binding-to-data.md#array-of-arrays), `prop` is the same as `col`.
+    ```js
+    const hot = new Handsontable(container, {
+      // the `cells` option set to a function
+      cells: function(row, col, prop) {
```
Comment (swistach) [inline docs/next/guides/getting-started/setting-options.md:547]:
```suggestion
      cells(row, col, prop) {
```

File: docs/next/guides/getting-started/setting-options.md [hunk 13]
```diff
@@ -1,239 +1,631 @@
...
+        // the `cells` function's body
+      }
+    });
+    ```
+3. In the [`cells`](@/api/options.md#cells) function's body, implement a logic that selects your required columns, rows, or cells (as combinations of `row` and `col` coordinates).<br>
+   For example:
+    ```js
+    const hot = new Handsontable(container, {
+      cells: function (row, col) {
+        if (row === 1 || row === 5 && col === 1) {
```
Comment (swistach) [inline docs/next/guides/getting-started/setting-options.md:557]:
```suggestion
      cells(row, col) {
        if ((row === 1 || row === 5) && col === 1) {
```

File: docs/next/guides/getting-started/setting-options.md [hunk 14]
```diff
@@ -1,239 +1,631 @@
...
+  licenseKey: 'non-commercial-and-evaluation',
+  data: Handsontable.helper.createSpreadsheetData(5, 10),
+  width: 'auto',
+  height: 'auto',
+  rowHeaders: true,
+  colHeaders: true,
+  // the `cells` option overwrites the top-level grid options
+  // apply only to cells selected by your custom logic
+  cells: function (row, col) {
+    if (row === 1 || row === 3 && col === 1) {
```
Comment (swistach) [inline docs/next/guides/getting-started/setting-options.md:588]:
```suggestion
  cells(row, col) {
    if ((row === 1 || row === 3) && col === 1) {
```

File: docs/next/guides/getting-started/setting-options.md [hunk 15]
```diff
@@ -248,30 +640,26 @@ const hot = new Handsontable(container, {
...
+  cell: [
+    {
+      // cell (0, 0) is read-only
+      row: 0,
+      col: 0,
+      readOnly: true,
+    },
+  ],
+  // the `cells` option's logic overwrites all other options
+  cells: function(row, column) {
```
Comment (swistach) [inline docs/next/guides/getting-started/setting-options.md:654]:
```suggestion
  cells(row, column) {
```

**Review by swistach — CHANGES_REQUESTED** [review]
What do you think to use `column` instead of `col`? Currently, we mix them in many places - maybe it's a good time to introduce consistent name convention in cell coordinates?

**Comment by kirszenbaum** [comment]
> Hey @kirszenbaum
> 
> Thanks for adding me as a reviewer. I added some suggestions within the code during the review. I also have some general thoughts
> 
> * we mention `setCellMeta`  after using it a couple of times before in examples
> * we do not mention that `setCellMeta`  does not generate the alteration of the `readOnly` mode till the cell is rerendered. Fortunately, `getCellMeta`  does not need a render so it's ok.
> * the `implementing custom logic` example seems to be pretty much the same as the one above. If we want to have this paragraph maybe we should use other options and/or play with conditions. Or maybe add some custom cellmeta (example https://jsfiddle.net/handsoncode/tb7rw0xk/)
> 
> Maybe I would also add something about the options which while added change the default set of keys in the dropdown and context menus. But it's only a thought. Maybe the context menu tutorial section is a better place.

File: docs/next/guides/getting-started/setting-options.md (hunk 10 above)
Comment (kirszenbaum) [reply docs/next/guides/getting-started/setting-options.md]: @mrpiotr-dev - done: 0efa67c

**Comment by kirszenbaum** [comment]
> What do you think to use `column` instead of `col`? Currently, we mix them in many places - maybe it's a good time to introduce consistent name convention in cell coordinates?

@mrpiotr-dev - to really keep it consistent, we'd have to make breaking changes in places like [this](https://handsontable.com/docs/api/options/#cell) and [this](https://handsontable.com/docs/api/options/#customborders)  (change `col` to `column`), right?

Overall, in options that require cell coordinates, `col` seems more prevalent than `column` to me, for example:
- https://handsontable.com/docs/api/options/#mergecells
- https://handsontable.com/docs/api/options/#comments

File: docs/next/guides/getting-started/setting-options.md [hunk 16]
```diff
@@ -1,239 +1,631 @@
...
+  // top-level grid options that apply to the entire grid
+  licenseKey: 'non-commercial-and-evaluation',
   data: Handsontable.helper.createSpreadsheetData(5, 10),
+  width: 'auto',
+  height: 'auto',
+  rowHeaders: true,
+  colHeaders: true,
+  // the `cells` options overwrite all other options
+  // apply only to each cell of rows 1 and 4, as specified in the function's body
   cells(row, column) {
```
Comment (mrpiotr-dev) [inline docs/next/guides/getting-started/setting-options.md:310]:
```suggestion
  cells(row) {
```

File: docs/next/guides/getting-started/setting-options.md [hunk 17]
```diff
@@ -248,30 +640,26 @@ const hot = new Handsontable(container, {
...
+  cell: [
+    {
+      // cell (0, 0) is read-only
+      row: 0,
+      col: 0,
+      readOnly: true,
+    },
+  ],
+  // the `cells` option's logic overwrites all other options
   cells(row, column) {
```
Comment (mrpiotr-dev) [inline docs/next/guides/getting-started/setting-options.md:654]:
```suggestion
  cells(row, col) {
```

File: docs/next/guides/getting-started/setting-options.md [hunk 18]
```diff
@@ -248,30 +640,26 @@ const hot = new Handsontable(container, {
...
-    if (row === 0 && column === 0) {
-      return {
-        readOnly: true,
-        className: 'bg-read-only',
-      }
-    }
-
+    // cell (2, 2) is editable
     if (row === 2 && column === 2) {
```
Comment (mrpiotr-dev) [inline docs/next/guides/getting-started/setting-options.md:656]:
```suggestion
    if (row === 2 && col === 2) {
```

**Review by mrpiotr-dev — CHANGES_REQUESTED** [review]
Indeed, you are right. Then it seems reasonable to use `col` over `column`. There are too many places where we use `col` (in source code) - changing them to `column` would be expensive in our clients' applications.

I request two changes that normalize `col` arguments and one removal of unnecessary/unused arguments.

**Review by mrpiotr-dev — APPROVED** [review]
LGTM :+1:

**Pull request merged**